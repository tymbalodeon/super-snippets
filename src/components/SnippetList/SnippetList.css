.SnippetList {
  margin-top: 50px;
}
