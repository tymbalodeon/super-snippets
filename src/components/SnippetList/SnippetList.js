import React, { Component } from 'react';
import Snippet from '../Snippet/Snippet';
import './SnippetList.css';
export default class List extends Component {
  renderSnippets() {
    const { snippets, project_id, updateSnippetId } = this.props;
    let finalList = snippets;
    if (project_id) {
      finalList = snippets.filter(snippet => snippet.project_id === project_id);
    }
    return finalList.map(snippet => (
      <Snippet
        key={snippet.id}
        snippet={snippet}
        updateSnippetId={updateSnippetId}
      />
    ));
  }

  render() {
    const { error } = this.props;
    return (
      <div className="SnippetList">
        {error ? (
          <p className="red">There was an error, try again</p>
        ) : (
          this.renderSnippets()
        )}
      </div>
    );
  }
}
