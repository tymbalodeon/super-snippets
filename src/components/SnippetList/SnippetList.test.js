import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SnippetList from './SnippetList';

describe('<SnippetList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<SnippetList />, div);

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer.create(<SnippetList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
