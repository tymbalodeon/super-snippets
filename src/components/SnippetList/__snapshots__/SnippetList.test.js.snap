// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SnippetList /> renders the UI as expected 1`] = `""`;
